neo/Network/RPC/WalletAPI.cs

```diff
@@ -0,0 +1,190 @@
+        public WalletAPI(RpcClient rpc)
+        {
+            rpcClient = rpc;
+            nep5API = new Nep5API(rpc);
+        }
+
+        /// <summary>
+        /// Get unclaimed gas with address, scripthash or public key string
+        /// </summary>
+        /// <param name="addressOrHash">address, scripthash or public key string
+        /// Example: address ("AV556nYUwyJKNv8Xy7hVMLQnkmKPukw6x5"), scripthash ("0x6a38cd693b615aea24dd00de12a9f5836844da91"), public key ("02f9ec1fd0a98796cf75b586772a4ddd41a0af07a1dbdf86a7238f74fb72503575")</param>
+        /// <returns></returns>
+        public decimal GetUnclaimedGas(string addressOrHash)
+        {
+            UInt160 account = addressOrHash.ToUInt160();
+            return GetUnclaimedGas(account);
+        }
+
+        /// <summary>
+        /// Get unclaimed gas
+        /// </summary>
+        /// <param name="account">account scripthash</param>
+        /// <returns></returns>
+        public decimal GetUnclaimedGas(UInt160 account)
+        {
+            UInt160 scriptHash = NativeContract.NEO.Hash;
+            BigInteger balance = nep5API.TestInvoke(scriptHash, "unclaimedGas", account, rpcClient.GetBlockCount() - 1)
+                .Stack.Single().ToStackItem().GetBigInteger();
+            return ((decimal)balance) / (long)NativeContract.GAS.Factor;
+        }
+
+        /// <summary>
+        /// Get Neo Balance
+        /// </summary>
+        /// <param name="addressOrHash">address, scripthash or public key string
+        /// Example: address ("AV556nYUwyJKNv8Xy7hVMLQnkmKPukw6x5"), scripthash ("0x6a38cd693b615aea24dd00de12a9f5836844da91"), public key ("02f9ec1fd0a98796cf75b586772a4ddd41a0af07a1dbdf86a7238f74fb72503575")</param>
+        /// <returns></returns>
+        public uint GetNeoBalance(string addressOrHash)
+        {
+            BigInteger balance = GetTokenBalance(NativeContract.NEO.Hash.ToString(), addressOrHash);
+            return (uint)balance;
+        }
+
+        /// <summary>
+        /// Get Gas Balance
+        /// </summary>
+        /// <param name="addressOrHash">address, scripthash or public key string
+        /// Example: address ("AV556nYUwyJKNv8Xy7hVMLQnkmKPukw6x5"), scripthash ("0x6a38cd693b615aea24dd00de12a9f5836844da91"), public key ("02f9ec1fd0a98796cf75b586772a4ddd41a0af07a1dbdf86a7238f74fb72503575")</param>
+        /// <returns></returns>
+        public decimal GetGasBalance(string addressOrHash)
+        {
+            BigInteger balance = GetTokenBalance(NativeContract.GAS.Hash.ToString(), addressOrHash);
+            return ((decimal)balance) / (long)NativeContract.GAS.Factor;
+        }
+
+        /// <summary>
+        /// Get token balance with string parameters
+        /// </summary>
+        /// <param name="tokenHash">token script hash, Example: "0x43cf98eddbe047e198a3e5d57006311442a0ca15"(NEO)</param>
+        /// <param name="addressOrHash">address, scripthash or public key string
```

Comment:
maybe refactor `addressOrHash` to something else that also gives the idea of public key.

Reply:
Good, renamed to `account`.